tools/export/gnuarmeclipse/__init__.py

```diff
@@ -0,0 +1,890 @@
+        -data /path/to/workspace \
+        -import /path/to/project \
+        -cleanBuild "project[/configuration] | all"
+        """
+
+        # TODO: possibly use the log file.
+
+        # Create a temporary folder for the workspace.
+        tmp_folder = tempfile.mkdtemp()
+
+        cmd = [
+            'eclipse', 
+            '--launcher.suppressErrors',
+            '-nosplash',
+            '-application org.eclipse.cdt.managedbuilder.core.headlessbuild',
+            '-data', tmp_folder,
+            '-import', os.getcwd(),
+            '-cleanBuild', 'all',
+        ]
+
+        ret_code = subprocess.call(cmd)
+
+        # Cleanup the exported and built files
+        if cleanup:
+            os.remove(log_name)
+            os.remove('.project')
+            os.remove('.cproject')
+            if exists('Debug'):
+                shutil.rmtree('Debug')
+            if exists('Release'):
+                shutil.rmtree('Release')
+
+        # Always remove the temporary folder.
+        if exists(tmp_folder):
+            shutil.rmtree(tmp_folder)
+
+        if ret_code == 0:
+            # Return Success
+            return 0
+
+        # Seems like something went wrong.
+        return -1
+
+    # -------------------------------------------------------------------------
+    # Process source files/folders exclusions.
+
+    def compute_exclusions(self):
+        """
+        With the project root as the only source folder known to CDT,
+        based on the list of source files, compute the folders to not
+        be included in the build.
+
+        The steps are:
+        - get the list of source folders, as dirname(source_file)
+        - compute the top folders (subfolders of the project folder)
+        - iterate all subfolders and add them to a tree, with all 
+        nodes markes as 'not used'
+        - iterate the source folders and mark them as 'used' in the
+        tree, including all intermediate nodes
+        - recurse the tree and collect all unused folders; descend
+        the hierarchy only for used nodes
+        """
+        source_folders = [self.filter_dot(s) for s in set(dirname(
+            src) for src in self.resources.c_sources + self.resources.cpp_sources + self.resources.s_sources)]
+        if '.' in source_folders:
+            source_folders.remove('.')
+
+        # print 'source folders'
+        # print source_folders
+
+        top_folders = [f for f in set(s.split(os.sep)[0]
+                                      for s in source_folders)]
+        # print 'top folders'
+        # print top_folders
+
+        self.source_tree = {}
+        for top_folder in top_folders:
+            for root, dirs, files in os.walk(top_folder):
+                if len(dirs) == 0:
+                    self.add_source_folder_to_tree(root)
+
+        for folder in source_folders:
+            self.add_source_folder_to_tree(folder, True)
+
+        # print
```

Comment:
shall these (all # print here) be removed (I assume they were used during the development) ?